protocols/kad/src/behaviour.rs

```diff
@@ -802,6 +909,14 @@ pub enum KademliaOut {
         /// List of peers ordered from closest to furthest away.
         closer_peers: Vec<PeerId>,
     },
+
+    /// Result of a `GET_VALUE` query
+    GetValueRes {
+        /// The result that we have probably received
+        result: Option<(Multihash, Vec<u8>)>,
```

Comment:
Is the key really needed in the result, i.e. a tuple needed? Is the key even included in the response? The libp2p-specs are a bit vague, but it only mentions that the response has `record` (in the protobuf) set if the value is found.

Reply:
My thoughts on this: we most likely need to know to what key request we received this response, since, `Multihash` might(?) be different hashing algorithms or even something like `/pk/{bytes}` (?). In any case, my intuition is that we need the key here, as it happens with `GetProvidersResult`.

Reply:
It looks like it's checked [in the go implementation](https://github.com/libp2p/go-libp2p-kad-dht/blob/cf9bd9b649fb7df4a5632357a17d86d5ccc28649/dht.go#L200)